sched/sched/sched_timerexpiration.c

```diff
@@ -233,14 +231,23 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
 
   /* Returning zero means that there is no interesting event to be timed */
 
-#ifdef KEEP_ALIVE_HACK
   if (ret == 0)
     {
       /* Apply the keep alive hack */
 
-      return KEEP_ALIVE_TICKS;
+      /* The idle task always lies at the end of the task list,
+       * idle task no need time slice.
+       */
+
+      if (ntcb->flink != NULL)
```

Comment:
change to sched_idletask

Reply:
sched_idletask just check current cpu running task, nxsched_cpu_scheduler must workable if SMP on. I think @pkarashchenko means to use the similar logic to check idle task?

Reply:
Yes, you are right.